game-core/src/main/java/games/strategy/triplea/ui/unit/scroller/UnitScroller.java

```diff
@@ -0,0 +1,278 @@
+    panel.add(skipButton, BorderLayout.SOUTH);
+    panel.add(Box.createVerticalStrut(3));
+
+    panel.add(SwingComponents.leftBox(movesLeftLabel));
+
+    panel.add(Box.createVerticalStrut(5));
+    return panel;
+  }
+
+  /** Centers the map on the current territory shown in the unit scroller. */
+  public void centerOnCurrentMovableUnit() {
+    if (lastFocusedTerritory != null) {
+      mapPanel.setUnitHighlight(
+          Collections.singleton(
+              UnitScrollerModel.getMoveableUnits(
+                  lastFocusedTerritory,
+                  movePhaseSupplier.get(),
+                  currentPlayerSupplier.get(),
+                  skippedUnits)));
+      mapPanel.centerOnTerritoryIgnoringMapLock(lastFocusedTerritory);
+    } else {
+      centerOnNextMovableUnit();
+    }
+  }
+
+  /**
+   * Skips the units in the current territory. Scroller will not scroll back to these units and the
+   * units will not be highlighted.
+   */
+  public void skipCurrentUnits() {
+    if (lastFocusedTerritory != null) {
+      skippedUnits.addAll(lastFocusedTerritory.getUnits());
+      updateMovesLeftLabel();
+    }
+    centerOnNextMovableUnit();
+  }
+
+  public void centerOnNextMovableUnit() {
+    centerOnMovableUnit(true);
+  }
+
+  public void centerOnPreviousMovableUnit() {
+    centerOnMovableUnit(false);
+  }
+
+  private void centerOnMovableUnit(final boolean selectNext) {
+    final List<Territory> allTerritories;
+    gameData.acquireReadLock();
+    try {
+      allTerritories = new ArrayList<>(gameData.getMap().getTerritories());
+    } finally {
+      gameData.releaseReadLock();
+    }
+
+    final int size = allTerritories.size();
+
+    if (!selectNext) {
+      Collections.reverse(allTerritories);
+    }
+    // new focused index is 1 greater
+    int newFocusedIndex =
+        lastFocusedTerritory == null ? 0 : allTerritories.indexOf(lastFocusedTerritory) + 1;
+    if (newFocusedIndex >= size) {
+      // if we are larger than the number of territories, we must start back at zero
+      newFocusedIndex = 0;
+    }
+    Territory newFocusedTerritory = null;
+    // make sure we go through every single territory on the board
+    int i = 0;
+    while (i < size) {
+      final Territory t = allTerritories.get(newFocusedIndex);
+      final List<Unit> matchedUnits =
+          UnitScrollerModel.getMoveableUnits(
+              t, movePhaseSupplier.get(), currentPlayerSupplier.get(), skippedUnits);
+
+      if (!matchedUnits.isEmpty()) {
+        drawUnitAvatarPane(t);
+        newFocusedTerritory = t;
+        final Collection<List<Unit>> highlight = new ArrayList<>();
+        highlight.add(matchedUnits);
+        mapPanel.setUnitHighlight(highlight);
+        break;
+      }
+      // make sure to cycle through the front half of territories
+      if ((newFocusedIndex + 1) >= size) {
+        newFocusedIndex = 0;
+      } else {
+        newFocusedIndex++;
+      }
```

Comment:
```java
newFocusedIndex = newFocusedIndex + 1 % allTerritories.size()
```

Reply:
I think you're right; my risk tolerance is not enough to make this update right now. I really think it's right, but adding a bug in here is not worth the small simplification, I'm being very conservative on this.

Reply:
Fair enough